Lib/test/test_pyrepl/test_interact.py

```diff
@@ -273,3 +273,28 @@ def test_incomplete_statement(self):
         code = "if foo:"
         console = InteractiveColoredConsole(namespace, filename="<stdin>")
         self.assertTrue(_more_lines(console, code))
+
+
+class TestWarnings(unittest.TestCase):
+    def test_pep_765_warning(self):
+        """
+        Test that a SyntaxWarning emitted from the
+        AST optimizer is only shown once in the REPL.
+        """
+        # gh-131927
+        console = InteractiveColoredConsole()
+        code = dedent("""\
+        def f():
+            try:
+                return 1
+            finally:
+                return 2
+        """)
+
+        f = io.StringIO()
+        with contextlib.redirect_stderr(f):
+            result = console.runsource(code)
+        self.assertFalse(result)
+        self.assertEqual(f.getvalue(),
+                         "<input>:5: SyntaxWarning: "
+                         "'return' in a 'finally' block\n")
```

Comment:
This test will fail if we add another unrelated warning. Maybe just check that this string appears exactly once in `f.getvalue()`?